extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/WrongAnnotationUsageProcessor.java

```diff
@@ -72,35 +75,72 @@ void detect(ArcConfig config, ApplicationIndexBuildItem applicationIndex, Custom
             NestingType nestingType = clazz.nestingType();
             if (NestingType.ANONYMOUS == nestingType || NestingType.LOCAL == nestingType
                     || (NestingType.INNER == nestingType && !Modifier.isStatic(clazz.flags()))) {
-                // Annotations declared on the class, incl. the annotations added via transformers
-                Collection<AnnotationInstance> classAnnotations = transformedAnnotations.getAnnotations(clazz);
-                if (classAnnotations.isEmpty() && clazz.annotationsMap().isEmpty()) {
-                    continue;
-                }
-                if (scopeAnnotations.isScopeIn(classAnnotations)) {
+                // Annotations declared on the class level, incl. the annotations added via transformers
+                List<AnnotationInstance> classLevelAnnotations = transformedAnnotations.getAnnotations(clazz).stream()
```

Comment:
There is no need for filtering - `transformedAnnotations.getAnnotations(clazz)` only returns the annotation instances declared on the class.

Reply:
Good point, I was too paranoid here